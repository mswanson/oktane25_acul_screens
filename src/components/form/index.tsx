export { ULThemeFloatingLabelField } from "./ULThemeFloatingLabelField";
export { ULThemeFormMessage } from "./ULThemeFormMessage";
