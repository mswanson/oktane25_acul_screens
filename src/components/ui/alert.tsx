import * as React from "react";

import { cva, type VariantProps } from "class-variance-authority";

import { cn } from "@/lib/utils";

const alertVariants = cva(
  "shadow-input-hover bg-input relative grid w-full grid-cols-[0_1fr] items-start gap-y-0.5 overflow-clip rounded-3xl p-3 text-sm has-[>svg]:grid-cols-[calc(var(--spacing)*4)_1fr] has-[>svg]:gap-x-3 [&>svg]:size-4 [&>svg]:translate-y-0.5 [&>svg]:text-current",
  {
    variants: {
      variant: {
        default: "text-foreground border-b-2 border-transparent",
        info: "text-info-foreground [&>svg]:text-info-foreground",
        success: "text-success-foreground [&>svg]:text-success-foreground",
        warning: "text-warning-foreground [&>svg]:text-warning-foreground",
        destructive:
          "text-destructive-foreground [&>svg]:text-destructive-foreground",
      },
    },
    defaultVariants: {
      variant: "default",
    },
  }
);

function Alert({
  className,
  variant,
  ...props
}: React.ComponentProps<"div"> & VariantProps<typeof alertVariants>) {
  return (
    <div
      data-slot="alert"
      role="alert"
      className={cn(alertVariants({ variant }), className)}
      {...props}
    />
  );
}

function AlertTitle({ className, ...props }: React.ComponentProps<"div">) {
  return (
    <div
      data-slot="alert-title"
      className={cn(
        "col-start-2 line-clamp-1 flex h-auto min-h-4 leading-4.5 font-medium tracking-tight",
        className
      )}
      {...props}
    />
  );
}
function AlertDescription({
  className,
  ...props
}: React.ComponentProps<"div">) {
  return (
    <div
      data-slot="alert-description"
      className={cn(
        "text-muted-foreground col-start-2 grid justify-items-start gap-1 text-sm [&_p]:leading-relaxed",
        className
      )}
      {...props}
    />
  );
}

function AlertAction({ className, ...props }: React.ComponentProps<"div">) {
  return (
    <div
      className={cn(
        "absolute top-2 right-2 flex items-center space-x-4",
        className
      )}
      {...props}
    />
  );
}

export { Alert, AlertAction, AlertDescription, AlertTitle };
