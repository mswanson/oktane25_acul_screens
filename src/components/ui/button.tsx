"use client";

import { useRender } from "@base-ui-components/react/use-render";
import { cva, type VariantProps } from "class-variance-authority";

import { cn } from "@/lib/utils";

const buttonVariants = cva(
  "focus-visible:ring-ring aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive theme-default:active:scale-[0.99] relative box-border inline-flex shrink-0 items-center justify-center gap-2 overflow-hidden text-sm font-medium whitespace-nowrap transition-all duration-150 ease-in-out outline-none focus-visible:ring-4 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:shrink-0 [&_svg:not([class*='size-'])]:size-4",
  {
    variants: {
      variant: {
        primary:
          "shadow-button-resting hover:shadow-button-hover hover:border-primary/50 border-primary bg-primary text-primary-foreground hover:bg-primary/90 theme-default:before:from-primary-foreground/0 theme-default:before:to-primary-foreground/15 theme-default:before:absolute theme-default:before:top-0 theme-default:before:left-0 theme-default:before:block theme-default:before:h-full theme-default:before:w-full theme-default:before:bg-gradient-to-t theme-default:before:content-[''] border",
        outline:
          "dark:bg-muted/50 hover:text-accent-foreground shadow-button-outlined-resting hover:shadow-button-outlined-hover hover:border-accent bg-background hover:bg-muted text-primary border-primary/35 theme-default:before:from-primary/5 theme-default:before:to-primary/0 theme-default:before:absolute theme-default:before:top-0 theme-default:before:left-0 theme-default:before:block theme-default:before:h-full theme-default:before:w-full theme-default:before:bg-gradient-to-t theme-default:before:content-[''] border",
        ghost: "hover:bg-muted text-primary bg-transparent",
        destructive:
          "bg-destructive text-destructive-foreground hover:bg-destructive/90 shadow-button-destructive-resting hover:shadow-button-destructive-hover border-destructive-border/25 hover:border-destructive-border/50 theme-default:before:to-primary-foreground/50 theme-default:before:absolute theme-default:before:top-0 theme-default:before:left-0 theme-default:before:block theme-default:before:h-full theme-default:before:w-full theme-default:before:bg-gradient-to-t theme-default:before:content-[''] theme-default:border",
        link: "text-foreground underline-offset-4 hover:underline",
      },
      size: {
        default: "h-10 rounded-2xl px-4 py-2.5 has-[>svg]:px-3",
        xs: "h-7 rounded-md px-2 py-1.5 text-xs has-[>svg]:px-2",
        sm: "h-8 gap-1.5 rounded-xl px-3 py-2 text-xs has-[>svg]:px-2.5",
        lg: "h-12 rounded-3xl px-6 py-3 text-base has-[>svg]:px-4",
        icon: "size-7 rounded-xl",
      },
    },
    defaultVariants: {
      variant: "primary",
      size: "default",
    },
  }
);

export interface ButtonProps
  extends useRender.ComponentProps<"button">,
    VariantProps<typeof buttonVariants> {}

function Button({
  className,
  variant,
  size,
  render = <button />,
  ...props
}: ButtonProps) {
  const defaultProps: useRender.ElementProps<"button"> = {
    className: cn(buttonVariants({ variant, size }), className),
  };

  return useRender({
    render,
    props: { ...defaultProps, ...props },
  });
}

export { Button, buttonVariants };
