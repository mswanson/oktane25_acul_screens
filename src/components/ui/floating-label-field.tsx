import * as React from "react";

import { Field as FieldPrimitive } from "@base-ui-components/react";
import { cva, VariantProps } from "class-variance-authority";

import { cn } from "@/lib/utils";

const floatingLabelFieldVariants = cva(
  "bg-input aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive theme-default:active:scale-[0.99] relative box-border inline-flex w-full shrink-0 cursor-text items-center justify-center text-sm transition-[color,box-shadow] duration-150 ease-in-out outline-none disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:shrink-0 [&_svg:not([class*='size-'])]:size-4",
  {
    variants: {
      variant: {
        default:
          "text-input-foreground shadow-input-resting hover:shadow-input-hover focus-within:ring-ring focus-within:ring-4",
        error:
          "text-destructive-foreground shadow-input-destructive-resting hover:shadow-input-destructive-hover focus-within:ring-destructive-border/15 focus-within:ring-4",
      },
      size: {
        default: "h-14 rounded-2xl",
        sm: "h-12 rounded-2xl",
        lg: "h-16 rounded-2xl",
      },
    },
    defaultVariants: {
      variant: "default",
      size: "default",
    },
  }
);

export interface FloatingLabelFieldProps
  extends Omit<React.InputHTMLAttributes<HTMLInputElement>, "size"> {
  label: string;
  error?: boolean;
  helperText?: string;
  size?: VariantProps<typeof floatingLabelFieldVariants>["size"];
  variant?: VariantProps<typeof floatingLabelFieldVariants>["variant"];
  startAdornment?: React.ReactNode;
  endAdornment?: React.ReactNode;
}

const FloatingLabelField = React.forwardRef<
  HTMLInputElement,
  FloatingLabelFieldProps
>(
  (
    {
      className,
      variant,
      size,
      error,
      helperText,
      label,
      startAdornment,
      endAdornment,
      ...props
    },
    ref
  ) => {
    const [focused, setFocused] = React.useState(false);
    const [hasValue, setHasValue] = React.useState(
      Boolean(props.value || props.defaultValue)
    );
    const isDisabled = props.disabled;

    const handleFocus = (e: React.FocusEvent<HTMLInputElement>) => {
      setFocused(true);
      props.onFocus?.(e);
    };

    const handleBlur = (e: React.FocusEvent<HTMLInputElement>) => {
      setFocused(false);
      setHasValue(Boolean(e.target.value));
      props.onBlur?.(e);
    };

    const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
      setHasValue(Boolean(e.target.value));
      props.onChange?.(e);
    };

    const isLabelFloating = focused || hasValue;

    return (
      <div>
        <FieldPrimitive.Root
          className={cn(
            floatingLabelFieldVariants({
              variant: error ? "error" : variant,
              size,
            }),
            "group relative items-end gap-0.5",
            isDisabled &&
              "bg-input-muted text-input-foreground cursor-not-allowed opacity-50",
            isDisabled && variant === "default" && "bg-input-muted",
            startAdornment && "pl-[5px]",
            endAdornment && "pr-[5px]",
            className
          )}
        >
          <FieldPrimitive.Label
            htmlFor={props.id}
            className={cn(
              "pointer-events-none absolute top-1/2 left-4 z-[10] -translate-y-1/2 text-sm transition-all duration-150 ease-in-out origin-left",
              startAdornment && "left-0",
              // Floating state - position within border area
              isLabelFloating &&
                "scale-80 -translate-y-[1.18rem] top-2 z-10 bg-input px-1",
              error ? "text-destructive-foreground" : "text-muted-foreground",
              focused && !error && "text-primary"
            )}
          >
            {label}
          </FieldPrimitive.Label>
          {startAdornment && (
            <div className="flex h-full items-center justify-center [&_svg]:pointer-events-none [&_svg]:shrink-0 [&_svg:not([class*='size-'])]:size-4">
              {startAdornment}
            </div>
          )}
          <div className="relative flex-1">
            <FieldPrimitive.Control
              id={props.id}
              className={cn(
                "h-14 w-full flex-1 rounded-2xl bg-transparent px-3 py-4 outline-none file:border-0 file:bg-transparent file:text-sm file:font-medium",
                // Override browser autofill styles without blocking borders
                "autofill:bg-transparent",
                "autofill:[-webkit-text-fill-color:inherit]",
                "autofill:[transition:background-color_9999s_ease-in-out_0s]",
                isDisabled &&
                  "bg-input-muted text-input-foreground cursor-not-allowed opacity-50",
                startAdornment ? "rounded-l-none pl-0" : "pl-3",
                endAdornment ? "rounded-r-none pr-0" : "pr-3",
                size === "sm" && "h-12 py-3 text-sm",
                size === "lg" && "h-16 py-5 text-base"
              )}
              // Spread props first, then override with our handlers
              // This ensures our focus/blur handlers take precedence over React Hook Form's
              // allowing proper floating label state management
              {...props}
              ref={ref}
              onFocus={handleFocus}
              onBlur={handleBlur}
              onChange={handleChange}
            />
          </div>
          {endAdornment && (
            <div className="flex h-full items-center justify-center [&_svg]:pointer-events-none [&_svg]:shrink-0 [&_svg:not([class*='size-'])]:size-4">
              {endAdornment}
            </div>
          )}
        </FieldPrimitive.Root>
        {helperText && (
          <p
            className={cn(
              "mt-1.5 text-xs",
              error ? "text-destructive-foreground" : "text-muted-foreground"
            )}
          >
            {helperText}
          </p>
        )}
      </div>
    );
  }
);

FloatingLabelField.displayName = "FloatingLabelField";

export { FloatingLabelField, floatingLabelFieldVariants };
