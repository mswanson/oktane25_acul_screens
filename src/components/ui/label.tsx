"use client";

import * as React from "react";

import { Field as FieldPrimitive } from "@base-ui-components/react";

import { cn } from "@/lib/utils";

function Label({
  className,
  ...props
}: React.ComponentProps<typeof FieldPrimitive.Label>) {
  return (
    <FieldPrimitive.Root>
      <FieldPrimitive.Label
        data-slot="label"
        className={cn(
          "flex items-center gap-2 text-sm leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50",
          className
        )}
        {...props}
      />
    </FieldPrimitive.Root>
  );
}
export { Label };
