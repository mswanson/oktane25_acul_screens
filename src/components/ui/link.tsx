import * as React from "react";

import { cva, VariantProps } from "class-variance-authority";

import { cn } from "@/lib/utils";

const linkVariants = cva(
  "focus-within:ring-ring inline-flex items-center gap-2 rounded-md py-0.5 underline-offset-4 transition-colors focus:ring-3 focus-visible:outline-hidden",
  {
    variants: {
      variant: {
        destructive: "text-destructive hover:text-destructive/90",
        muted: "text-muted hover:text-muted/80",
        primary: "text-primary hover:text-primary/90",
      },
      underline: {
        none: "no-underline",
        hover: "no-underline hover:underline",
        always: "underline",
      },
    },
    defaultVariants: {
      variant: "primary",
      underline: "always",
    },
  }
);

export interface LinkProps
  extends React.AnchorHTMLAttributes<HTMLAnchorElement>,
    VariantProps<typeof linkVariants> {
  /**
   * Optional ref for the link element.
   */
  ref?: React.Ref<HTMLAnchorElement>;
}

function Link({
  className,
  children,
  variant,
  underline,
  ref,
  ...props
}: LinkProps) {
  return (
    <a
      ref={ref}
      className={cn(linkVariants({ variant, underline }), className)}
      {...props}
    >
      {children}
    </a>
  );
}

export { Link };
