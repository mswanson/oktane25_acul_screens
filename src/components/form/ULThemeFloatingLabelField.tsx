import { cva } from "class-variance-authority";

import {
  FloatingLabelField as BaseFloatingLabelField,
  type FloatingLabelFieldProps as BaseFloatingLabelFieldProps,
} from "@/components/ui/floating-label-field";
import { useFormField } from "@/components/ui/form";
import { cn } from "@/lib/utils";

const ulThemeFloatingLabelFieldVariants = cva(
  [
    // Layout & Spacing
    "mb-2",

    // Base Background & Colors
    "theme-universal:bg-input-bg",
    "theme-universal:text-input-text",

    // Border Styling
    "theme-universal:border-(length:--ul-theme-border-input-border-weight)",
    "theme-universal:border-input-border",

    // Border Radius
    "theme-universal:rounded-input",

    // Typography - Input Text
    "theme-universal:text-(length:--ul-theme-font-body-text-size)",
    "theme-universal:font-body-text",

    // Placeholder Styling
    "theme-universal:placeholder:text-input-labels",
    "theme-universal:[&_input]:placeholder:text-input-labels",
    "theme-universal:placeholder:text-(length:--ul-theme-font-input-labels-size)",
    "theme-universal:[&_input]:placeholder:text-(length:--ul-theme-font-input-labels-size)",
    "theme-universal:placeholder:font-input-label",
    "theme-universal:[&_input]:placeholder:font-input-label",

    // Floating Label Background
    "theme-universal:[&_label]:bg-input-bg",

    // Floating Label - Base Styling (applies to all labels)
    "theme-universal:[&_label]:text-input-labels",
    "theme-universal:[&_label]:text-(length:--ul-theme-font-input-labels-size)",
    "theme-universal:[&_label]:font-input-label",
  ],
  {
    variants: {
      themeState: {
        default: [
          // Focus States - Default
          "theme-universal:focus-within:border-base-focus",
          "theme-universal:focus-within:ring-0",

          // Focus Label States - Override label color when focused
          "theme-universal:focus-within:[&_label]:text-base-focus",
        ],
        error: [
          // Error States
          "theme-universal:text-error",
          "theme-universal:border-error",
          "theme-universal:focus-within:border-error",
          "theme-universal:focus-within:ring-0",

          // Error Label States - Force override using color CSS property
          "theme-universal:[&_label]:text-error",
        ],
      },
    },
    defaultVariants: {
      themeState: "default",
    },
  }
);

export interface ULThemeFloatingLabelFieldProps
  extends BaseFloatingLabelFieldProps {
  /**
   * Additional wrapper class for form field container
   */
  wrapperClassName?: string;
}

function ULThemeFloatingLabelField({
  className,
  variant = "default",
  size = "default",
  wrapperClassName,
  error,
  ...props
}: ULThemeFloatingLabelFieldProps) {
  // Get the form field context for proper ID association
  const { formItemId } = useFormField();

  // Determine variant based on error prop
  const effectiveVariant = error ? "error" : variant;

  // Generate theme classes using CVA - automatically use error state when error prop is true
  const themeOverrides = ulThemeFloatingLabelFieldVariants({
    themeState: error ? "error" : "default",
  });

  return (
    <div className={cn("w-full", wrapperClassName)}>
      <BaseFloatingLabelField
        id={formItemId}
        className={cn(className, themeOverrides)}
        variant={effectiveVariant}
        size={size}
        error={error}
        {...props}
      />
    </div>
  );
}

ULThemeFloatingLabelField.displayName = "ULThemeFloatingLabelField";

export { ULThemeFloatingLabelField };
