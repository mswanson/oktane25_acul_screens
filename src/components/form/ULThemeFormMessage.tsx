import { AlertCircle } from "lucide-react";

import { FormMessage } from "@/components/ui/form";
import { cn } from "@/lib/utils";

export interface ULThemeFormMessageProps {
  /**
   * SDK error message (takes precedence over form validation errors)
   */
  sdkError?: string;
  /**
   * Whether there is a form validation error
   */
  hasFormError?: boolean;
  /**
   * Whether to show the error icon
   */
  showIcon?: boolean;
  /**
   * Additional CSS classes
   */
  className?: string;
}

const ULThemeFormMessage = ({
  sdkError,
  hasFormError,
  showIcon = true,
  className,
}: ULThemeFormMessageProps) => {
  // Don't render if no errors at all
  if (!sdkError && !hasFormError) {
    return null;
  }

  // Always render with consistent theming and icon for ANY error
  return (
    <div
      className={cn(
        "flex mb-2 items-center text-sm font-medium theme-universal:text-error",
        className
      )}
      role="alert"
      aria-live="polite"
    >
      {showIcon && <AlertCircle className="h-4 w-4 mr-1 flex-shrink-0" />}
      {sdkError ? (
        <p className="text-destructive text-sm theme-universal:text-error">
          {sdkError}
        </p>
      ) : (
        <FormMessage className="theme-universal:text-error" />
      )}
    </div>
  );
};

ULThemeFormMessage.displayName = "ULThemeFormMessage";

export { ULThemeFormMessage };
