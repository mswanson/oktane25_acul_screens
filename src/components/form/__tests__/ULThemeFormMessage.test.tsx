import React from "react";
import { useForm } from "react-hook-form";

import { render, screen } from "@testing-library/react";

import { Form, FormField, FormItem } from "@/components/ui/form";

import { ULThemeFormMessage } from "../ULThemeFormMessage";

// Test wrapper that mimics IdentifierForm usage pattern
const TestFormMessageWrapper = ({
  children,
  hasValidationError = false,
  validationMessage = "This field is required",
}: {
  children: React.ReactNode;
  hasValidationError?: boolean;
  validationMessage?: string;
}) => {
  const form = useForm({
    defaultValues: { identifier: "" },
  });

  React.useEffect(() => {
    if (hasValidationError) {
      form.setError("identifier", { message: validationMessage });
    }
  }, [form, hasValidationError, validationMessage]);

  return (
    <Form {...form}>
      <form>
        <FormField
          control={form.control}
          name="identifier"
          render={() => <FormItem>{children}</FormItem>}
        />
      </form>
    </Form>
  );
};

describe("ULThemeFormMessage Component", () => {
  // Snapshot Tests - covering main usage scenarios
  it("matches snapshot with SDK error", () => {
    const { container } = render(
      <TestFormMessageWrapper>
        <ULThemeFormMessage sdkError="Invalid email format" />
      </TestFormMessageWrapper>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("matches snapshot with form validation error", () => {
    const { container } = render(
      <TestFormMessageWrapper hasValidationError>
        <ULThemeFormMessage hasFormError={true} />
      </TestFormMessageWrapper>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("matches snapshot with no errors", () => {
    const { container } = render(
      <TestFormMessageWrapper>
        <ULThemeFormMessage hasFormError={false} />
      </TestFormMessageWrapper>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  // Functional Tests - real-world scenarios from IdentifierForm
  it("prioritizes SDK error over form validation (like in IdentifierForm)", () => {
    render(
      <TestFormMessageWrapper
        hasValidationError
        validationMessage="This field is required"
      >
        <ULThemeFormMessage
          sdkError="Invalid email format"
          hasFormError={true}
        />
      </TestFormMessageWrapper>
    );

    expect(screen.getByText("Invalid email format")).toBeInTheDocument();
    expect(
      screen.queryByText("This field is required")
    ).not.toBeInTheDocument();
  });

  it("shows form validation error when no SDK error (required field scenario)", () => {
    render(
      <TestFormMessageWrapper
        hasValidationError
        validationMessage="This field is required"
      >
        <ULThemeFormMessage hasFormError={true} />
      </TestFormMessageWrapper>
    );

    expect(screen.getByText("This field is required")).toBeInTheDocument();
    expect(screen.getByRole("alert")).toBeInTheDocument();
  });

  it("shows maximum length validation error", () => {
    render(
      <TestFormMessageWrapper
        hasValidationError
        validationMessage="Maximum 100 characters allowed"
      >
        <ULThemeFormMessage hasFormError={true} />
      </TestFormMessageWrapper>
    );

    expect(
      screen.getByText("Maximum 100 characters allowed")
    ).toBeInTheDocument();
  });

  it("renders nothing when no errors (clean state)", () => {
    const { container } = render(
      <TestFormMessageWrapper>
        <ULThemeFormMessage hasFormError={false} />
      </TestFormMessageWrapper>
    );

    expect(container.querySelector('[role="alert"]')).not.toBeInTheDocument();
  });

  it("applies Auth0 Universal Login theme styling", () => {
    const { container } = render(
      <TestFormMessageWrapper>
        <ULThemeFormMessage sdkError="Theme test error" />
      </TestFormMessageWrapper>
    );

    const errorContainer = container.querySelector('[role="alert"]');
    expect(errorContainer).toHaveClass("theme-universal:text-error");
  });

  it("includes error icon for accessibility", () => {
    const { container } = render(
      <TestFormMessageWrapper>
        <ULThemeFormMessage sdkError="Error with icon" />
      </TestFormMessageWrapper>
    );

    const icon = container.querySelector("svg");
    expect(icon).toBeInTheDocument();
    expect(icon).toHaveClass("h-4", "w-4");
  });

  it("can hide icon when specified", () => {
    const { container } = render(
      <TestFormMessageWrapper>
        <ULThemeFormMessage sdkError="Error without icon" showIcon={false} />
      </TestFormMessageWrapper>
    );

    expect(container.querySelector("svg")).not.toBeInTheDocument();
  });
});
