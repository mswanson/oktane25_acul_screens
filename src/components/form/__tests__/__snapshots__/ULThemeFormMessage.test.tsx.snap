// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ULThemeFormMessage Component matches snapshot with SDK error 1`] = `
<form>
  <div
    class="grid"
    data-slot="form-item"
  >
    <div
      aria-live="polite"
      class="flex mb-2 items-center text-sm font-medium theme-universal:text-error"
      role="alert"
    >
      <svg
        aria-hidden="true"
        class="lucide lucide-circle-alert h-4 w-4 mr-1 flex-shrink-0"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="10"
        />
        <line
          x1="12"
          x2="12"
          y1="8"
          y2="12"
        />
        <line
          x1="12"
          x2="12.01"
          y1="16"
          y2="16"
        />
      </svg>
      <p
        class="text-destructive text-sm theme-universal:text-error"
      >
        Invalid email format
      </p>
    </div>
  </div>
</form>
`;

exports[`ULThemeFormMessage Component matches snapshot with form validation error 1`] = `
<form>
  <div
    class="grid"
    data-slot="form-item"
  >
    <div
      aria-live="polite"
      class="flex mb-2 items-center text-sm font-medium theme-universal:text-error"
      role="alert"
    >
      <svg
        aria-hidden="true"
        class="lucide lucide-circle-alert h-4 w-4 mr-1 flex-shrink-0"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="10"
        />
        <line
          x1="12"
          x2="12"
          y1="8"
          y2="12"
        />
        <line
          x1="12"
          x2="12.01"
          y1="16"
          y2="16"
        />
      </svg>
      <p
        class="text-destructive text-sm theme-universal:text-error"
        data-slot="form-message"
        id="«r1»-form-item-message"
      >
        This field is required
      </p>
    </div>
  </div>
</form>
`;

exports[`ULThemeFormMessage Component matches snapshot with no errors 1`] = `
<form>
  <div
    class="grid"
    data-slot="form-item"
  />
</form>
`;
